### PR TITLE
Add wait-job plugin

### PR DESCRIPTION
**Pull request by brianpursley**
Homepage: https://github.com/brianpursley/kubectl-wait-job

Description: Waits for a job to complete or fail

### CODE REVIEW

**Comment by ahmetb** [comment]
:robot: _Beep beep! I’m a robot speaking on behalf of @ahmetb._ :robot:

-----

Thanks for submitting your kubectl plugin to Krew!
One of the krew-index maintainers will review it soon. Note that the reviews for new plugin submissions may take a few days.

In the meanwhile, here are a few tips to make your plugin manifest better:
* Make sure your plugin follows the [best practices](https://krew.sigs.k8s.io/docs/developer-guide/develop/best-practices/).
* Eliminate redundant wording form `shortDescription` (it should be max 50 characters).
* Try to word wrap your `description` to 80-character lines (no usage examples, please).

Thanks for your patience!
/kind new-plugin

**Comment by ahmetb** [comment]
Can this be achieved with a one-liner with "kubectl wait"? If so, it might be good to use that instead and of developing a single-purpose/single-resource plugin.

**Comment by brianpursley** [comment]
> Can this be achieved with a one-liner with "kubectl wait"? If so, it might be good to use that instead and of developing a single-purpose/single-resource plugin.

No, not really.  kubectl doesn't have a way to wait for multiple statuses (Complete or Failed).

See: 
* https://github.com/kubernetes/kubectl/issues/1629
* https://github.com/kubernetes/kubectl/issues/1224
* https://stackoverflow.com/questions/55073453/wait-for-kubernetes-job-to-complete-on-either-failure-success-using-command-line

It would be a new feature in kubectl to allow this kind of waiting, but to help now, I wanted to create a plugin that can be used for this case where it is needed.  As far as I understand this situation is unique to jobs.

**Comment by brianpursley** [comment]
I understand though if it's not a good fit for the krew index.  It could also just be a non-krew plugin for people who need this special use case.

File: plugins/wait-job.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+apiVersion: krew.googlecontainertools.github.com/v1alpha2
+kind: Plugin
+metadata:
+  name: wait-job
+spec:
+  version: v0.1.0
+  homepage: https://github.com/brianpursley/kubectl-wait-job
+  shortDescription: Waits for a job to complete or fail
```
Comment (ahmetb) [inline plugins/wait-job.yaml:8]:
```suggestion
  shortDescription: Waits for a Job to complete or fail
```

File: plugins/wait-job.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  description: |
+    Waits for a job to complete or fail, exiting with an 
+    exit code of 0 if the job completes successfully, or
+    1 if the job fails.
```
Comment (ahmetb) [inline plugins/wait-job.yaml:12]:
```suggestion
    Waits for a Job to complete or fail, exiting with an 
    exit code of 0 if the Job completes successfully, or
    1 if the Job fails.
```

**Comment by ahmetb** [comment]
You make a valid point, this is a deficiency in kubectl today. I assumed it's straightforward as if there's a single condition to wait on.

**Comment by brianpursley** [comment]
Changes applied

**Comment by ahmetb** [comment]
/lgtm
/approve
Thanks!
Also, please consider setting up [Krew release automation](https://krew.sigs.k8s.io/docs/developer-guide/release/automating-updates/) which helps you skip manually making updates to your Krew manifests on each new version and send a pull request. It’s a GitHub workflow bot that can run every time you push a new tag. Those PRs are automatically approved.

**Pull request merged**